pyccel/ast/internals.py

```diff
@@ -55,88 +56,132 @@
 
 class PyccelArraySize(PyccelInternalFunction):
     """
-    Class representing a call to a function which would
-    return the shape of an object in a given dimension
+    Gets the total number of elements in an array.
+
+    Class representing a call to a function which would return
+    the total number of elements in a multi-dimensional array.
+
+    Parameters
+    ----------
+    arg : PyccelAstNode
+        An array of unknown size.
+    """
+    __slots__ = ()
+    name = 'size'
+
+    _dtype = NativeInteger()
+    _precision = -1
+    _rank  = 0
+    _shape = None
+    _order = None
+
+    def __init__(self, arg):
+        super().__init__(arg)
+
+    @property
+    def arg(self):
+        """ Object whose size is investigated.
+        """
+        return self._args[0]
+
+    def __str__(self):
+        return f'Size({self.arg})'
+
+    def __eq__(self, other):
+        if isinstance(other, PyccelArraySize):
+            return self.arg == other.arg
+        else:
```

Comment:
@yguclu Do you think we should ignore this? It is for comparing sizes. I think it was used to check if arrays were compatible but the handling of this regressed a little when we added functions returning arrays. I think it could be useful in the future and is simple enough to not be critical

Reply:
I am OK with ignoring this warning 😉

Reply:
/bot accept for reasons described above